test/e2e/multihoming_utils.go

```diff
@@ -29,24 +30,58 @@ func netCIDR(netCIDR string, netPrefixLengthPerNode int) string {
 	return fmt.Sprintf("%s/%d", netCIDR, netPrefixLengthPerNode)
 }
 
-// takes ipv4 and ipv6 cidrs and returns the correct type for the cluster under test
-func correctCIDRFamily(ipv4CIDR, ipv6CIDR string) string {
-	return strings.Join(selectCIDRs(ipv4CIDR, ipv6CIDR), ",")
+func joinCIDRs(cidrs ...string) string {
+	return strings.Join(cidrs, ",")
 }
 
-// takes ipv4 and ipv6 cidrs and returns the correct type for the cluster under test
-func selectCIDRs(ipv4CIDR, ipv6CIDR string) []string {
+func splitCIDRs(cidrs string) []string {
+	return strings.Split(cidrs, ",")
+}
+
+func filterJoinedCIDRsAndMerge(cs clientset.Interface, cidrs string) string {
+	return joinCIDRs(filterCIDRs(cs, splitCIDRs(cidrs)...)...)
+}
+
+func filterCIDRs(cs clientset.Interface, cidrs ...string) []string {
+	var ipv4CIDRs, ipv6CIDRs []string
+	for _, cidr := range cidrs {
+		if utilsnet.IsIPv6CIDRString(cidr) {
+			ipv6CIDRs = append(ipv6CIDRs, cidr)
+		} else {
+			ipv4CIDRs = append(ipv4CIDRs, cidr)
+		}
+	}
 	// dual stack cluster
-	if isIPv6Supported() && isIPv4Supported() {
-		return []string{ipv4CIDR, ipv6CIDR}
+	if isIPv6Supported(cs) && isIPv4Supported(cs) {
+		return append(ipv4CIDRs, ipv6CIDRs...)
 	}
 	// is an ipv6 only cluster
-	if isIPv6Supported() {
-		return []string{ipv6CIDR}
+	if isIPv6Supported(cs) {
+		return ipv6CIDRs
 	}
+	//ipv4 only cluster
+	return ipv4CIDRs
+}
```

Comment:
Now that we are at it we should do this to support ipv4, dualstack and ipv6 and simplify it a little
```golang
var filteredCIDRs []string
if isIPv4Supported(cs) {
   filteredCIDRs = append(filteredCIDRs, ipv4CIDRs...)
}

if isIPv6Supported(cs) {
   filteredCIDRs = append(filteredCIDRs, ipv6CIDRs...)
}

return filteredCIDRs.   
}
```